### DISCUSSION TITLE
Seeking Help With Templater's create_new function

### DISCUSSION THREAD

**Discussion by KennSugiyama** [SilentVoid13/Templater#423] [Help]
For the past couple of days, I've been struggling getting Templater's create_new function to place the newly created file in a specific folder.

**Scenario:**

- I'd like to use an existing template called "hideeHo"
- I'd like to name the file: "Woohoo"
- I don't want to open the new file.
- I'd like to save the newly created file in /folderA/subFolder1.

From the documentation, I know I need to use: `tp.file.create_new(template: TFile ⎮ string, filename?: string, open_new: boolean = false, folder?: TFolder)`. I'm pretty sure my knowledge gap is in not knowing how to use TFolder.

What I Have So Far (This works, but places the newly created file in the default directory)

```
let templateFile = await tp.file.find_tfile("hideeHo").basename  
await tp.file.create_new(templateFile, "WhooHoo", false) 
```
I've tried:

1) `await tp.file.create_new(templateFile, "WhooHoo", false,"folderA/subFolder1/")` which doesn't work because I'm trying to pass a string.

2) All sorts of weird things to replicate what I did with the tp.file.find_tfile method.

Any help would much be appreciated! Thanks!

**Comment by huenchen** [comment]
I have used it like this:
- having a prompt asking for the file Name and then use this name and put the folder in front of it (in this case called Books)

```
<%*
let qcFileName = await tp.system.prompt("Note Title")
titleName = qcFileName
await tp.file.move("/Books/" + titleName);
-%>
```

**Reply by AB1908** [reply]
I'd second this suggestion. However, you can obtain a TFolder by pointing `app.vault.getAbstractFileByPath` at a folder path.
